### ISSUE TITLE
[dired-collapse] does not update after collapsed file action (delete, rename, ...)

### ISSUE THREAD

**Issue by zellerin** [Fuco1/dired-hacks#93]
Hello,

When you delete a collapsed tree (`dired-flag-file-deletion` and `dired-do-flagged-delete`), the deleted entry does not disappear from the dired listing. SImilar for rename.

Without collapsing mode - or when the entry is not collapsed item, it does disappear when deletion is done.

As a side note, I would think that it would be more intuitive to delete whole collapsed directory structure, not just the leaf, but I suppose this is a matter of preference.

**Comment by Fuco1** [comment]
Yes, it only operates on the leaf.  I think we should leave it that way since that's consistent with how Emacs handles nested listings by default.  I'm not sure how simple or not it would be to make things consistently operate in the way you describe (most likely every command would have to be made aware that it can be operating on a collapsed structure).

So for now the "solution" is to turn off the collapse mode and delete the directory regularly.

**Comment by Fuco1** [comment]
After a couple months of using this extension I think the behaviour you propose is better, it annoys me more often than not that the whole path remains in place.

I think I would even like it by default on but I still am not sure how to implement this.  Maybe with some crazy advices :/
